root_orchestrator/cloud-scheduler/resource_management/cluster_operations.py

```diff
@@ -0,0 +1,39 @@
+import os
+from requests import get, exceptions
+
+RESOURCE_ABSTRACTOR_ADDR = f"http://{os.environ.get('RESOURCE_ABSTRACTOR_URL')}:{os.environ.get('RESOURCE_ABSTRACTOR_PORT')}"
+
+def get_resources(**kwargs):
+    request_address = RESOURCE_ABSTRACTOR_ADDR + '/api/v1/resources'
+    try:
+        response = get(request_address, params=kwargs)
+        return response.json()
+    except exceptions.RequestException as e:
+        print('Calling Resource Abstractor /api/v1/resources not successful.')
+    
+    return None
```

Comment:
I fear that this None can potentially result in some fatal error. After logging the error in case of exception, we can return an empty list of resources, I believe. This will then result in failed scheduling results to the orchestrator. We can then decide over there if and when to command a new reschedule.

Reply:
Same thing for get_resource_by_x

Reply:
Good call!

Reply:
But for single resource/item, shouldn't we return None if request fails? Also for the case `get_resource_by_id` the caller is checking if the result is None, so I suppose it should be fine, no?